torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -596,16 +603,20 @@ c10::optional<std::vector<int64_t>> GetValueFromListConstructNode(
       : c10::nullopt;
 }
 
-void ProcessReshapeNode(Node* n) {
+void ProcessReshapeNode(Node* n, int opset_version) {
   if (ConstantValueMap::HasValue(n->input(1)->debugName())) {
     auto shape_temp =
         ConstantValueMap::GetValueInto1DInt64Vector(n->input(1)->debugName());
     auto shape_vector_0 =
         ConstantValueMap::GetShapeInto1DInt64VectorWithOneUnknown(
             n->input(0)->debugName());
+    int allowzero = 1;
```

Comment:
for older opsets the behavior in onnx should be allowzero = 0 right?

Reply:
I don't this should affect shape inference as such, as the check is only valid for opset 14 onwards. I was wondering what would be the behavior if
1) node has no allowero
2) opset >= 14

Reply:
For the case you mentioned node has non allowzero and opset>=14, it should be same behavior as old opset, right?

Also I agree that `allowzero = 0` since this is the default value and back compatibility behavior for old opsets. It is true it does not affect shape inference, but seems more reasonable to set it to 0 by initialization.

Reply:
nit: Can we change `ComputeShapeFromReshape` API to pass `Node* n` as argument? In that case, we don't need initialize `allow_zero` here, but just do this `opset_version >= 14` case inside `ComputeShapeFromReshape`. Then we only have one place to handle   `opset_version >= 14` .

Reply:
Updated this section